src/arkode/arkode_arkstep_nls.c

```diff
@@ -75,25 +74,23 @@ int ARKStepSetNonlinearSolver(void *arkode_mem, SUNNonlinearSolver NLS)
   retval = SUNNonlinSolSetConvTestFn(step_mem->NLS, arkStep_NlsConvTest,
                                      (void*) ark_mem);
   if (retval != ARK_SUCCESS) {
-    arkProcessError(ark_mem, ARK_ILL_INPUT, "ARKODE::ARKStep",
-                    "ARKStepSetNonlinearSolver",
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
                     "Setting convergence test function failed");
     return(ARK_ILL_INPUT);
   }
 
   /* set default nonlinear iterations */
   retval = SUNNonlinSolSetMaxIters(step_mem->NLS, step_mem->maxcor);
```

Comment:
Can this code block be replaced by the simpler
```
SUNCheckCall(SUNNonlinSolSetMaxIters(step_mem->NLS, step_mem->maxcor));
```
?

Reply:
No, because it would break backwards compatibility which allows for a user to set an arkode error handler that `arkProcessError` would invoke. Or, we would have to somehow make the arkode error handler that already exists work with the `SUNErrHandlerFn` handlers.

Reply:
Given that I have never heard of anyone implementing a custom error handler for ARKODE, then I think that a future major release would include unification of error handling capabilities.  Specifically, the package-specific error handlers could be implemented as `SUNErrHandlerFn` objects, with package-specific user error handlers removed.  Then, when each integrator is created, it could attach its package-specific `SUNErrHandlerFn` within the linked list of error handlers.